tests/display/where_clauses.rs

```diff
@@ -0,0 +1,111 @@
+#[test]
+fn test_complicated_bounds() {
+    reparse_test!(
+        program {
+            struct Foo { }
+            trait Bar { }
+            trait Baz<T> { }
+            trait Bax<T> { type BaxT; }
+            trait Test {
+                type Assoc<T>: Bar + Baz<Foo> + Bax<T, BaxT=T>
+                    where
+                        Foo: Bax<T, BaxT=T>,
+                        Foo: Bar,
+                        dyn Bar: Baz<Foo>;
+            }
+        }
+        produces {
```

Comment:
Theoretically, if we *do* want to do this, why not just check that the program outputs itself?

Reply:
The problem here is that going through the loop always adds additional clauses to an `AliasEq` where clause. Lowering always transforms `Foo: Bax<T, BaxT=T>` into `Foo: Bax<T, BaxT=T>, Foo: Bax<T>`, and trying to undo that seemed like more work than it'd be worth.

Even if we included the output program as the source, it'd just add another `Foo: Bax<T>` clause and we'd be in the same place.

I'll add a comment to that to the test code, though. That's useful information that we didn't include!